packages/analytics-client-common/src/attribution/helpers.ts

```diff
@@ -27,25 +27,37 @@ export const isNewCampaign = (
   current: Campaign,
   previous: Campaign | undefined,
   options: Options,
+  logger: Logger,
   isNewSession = true,
 ) => {
   const { referrer, referring_domain, ...currentCampaign } = current;
   const { referrer: _previous_referrer, referring_domain: prevReferringDomain, ...previousCampaign } = previous || {};
 
   if (isExcludedReferrer(options.excludeReferrers, current.referring_domain)) {
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    logger.log(`This is not a new campaign because ${current.referring_domain} is in the exclude referrer list.`);
     return false;
   }
 
   //In the same session, direct traffic should not override or unset any persisting query params
   if (!isNewSession && isDirectTraffic(current) && previous) {
+    logger.log('This is not a new campaign because this is a direct traffic in the same session.');
     return false;
   }
 
   const hasNewCampaign = JSON.stringify(currentCampaign) !== JSON.stringify(previousCampaign);
   const hasNewDomain =
     domainWithoutSubdomain(referring_domain || '') !== domainWithoutSubdomain(prevReferringDomain || '');
 
-  return !previous || hasNewCampaign || hasNewDomain;
+  const result = !previous || hasNewCampaign || hasNewDomain;
+
+  if (!result) {
+    logger.log("This is not a new campaign because it's the same as the previous one.");
```

Comment:
Thanks for adding this! Should this be `logger.log` or `logger.debug`?

Reply:
Updated